inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -1456,586 +1108,709 @@ static void initScratchArea(const cv::GMatDesc& in, const Size& outSz,
+                    cv::gapi::fluid::Buffer& out4,
+                    cv::gapi::fluid::Buffer& scratch) {
+        constexpr int numChan = 4;
+        std::array<std::reference_wrapper<cv::gapi::fluid::Buffer>, numChan> out = {out1, out2, out3, out4};
+        calcRowLinearC<uint8_t, linear::Mapper, numChan>(in, out, scratch);
     }
 };
 
-GAPI_FLUID_KERNEL(FUpscalePlaneArea32f, UpscalePlaneArea32f, true) {
-    static const int Window = 1;
-    static const int LPI = 4;
-    static const auto Kind = cv::GFluidKernel::Kind::Resize;
+#if defined __GNUC__
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#endif
 
-    static void initScratch(const cv::GMatDesc& in,
-                            Size outSz, int /*interp*/,
-                            cv::gapi::fluid::Buffer &scratch) {
-        initScratchLinear<float, areaUpscale32f::Mapper>(in, outSz, scratch, 0);
-    }
+template<typename T, typename Mapper>
+static inline void calcAreaRow(const cv::gapi::fluid::View& in, cv::gapi::fluid::Buffer& out,
+                               cv::gapi::fluid::Buffer& scratch) {
+    using Unit = typename Mapper::Unit;
+    using alpha_type = typename Mapper::alpha_type;
+    using index_type = typename Mapper::index_type;
+    using  work_type = typename Mapper::work_type;
 
-    static void resetScratch(cv::gapi::fluid::Buffer& /*scratch*/) {
-    }
+    Size inSz  =  in.meta().size;
+    Size outSz = out.meta().size;
+
+    // this method is valid only for down-scale
+    GAPI_DbgAssert(inSz.width  >= outSz.width);
+    GAPI_DbgAssert(inSz.height >= outSz.height);
+
+//  Mapper xmapper(inSz.width,  outSz.width);
+    Mapper ymapper(inSz.height, outSz.height);
+
+    auto *xmaxdf = scratch.OutLine<const int>();
+    auto  maxdif = xmaxdf[0];
+
+    auto *xindex = reinterpret_cast<const index_type*>(xmaxdf + 1);
+    auto *xalpha = reinterpret_cast<const alpha_type*>(xindex + outSz.width);
+    auto *vbuf_c = reinterpret_cast<const  work_type*>(xalpha + outSz.width * maxdif);
+
+    auto *vbuf = const_cast<work_type*>(vbuf_c);
+
+    int iny = in.y();
+    int y = out.y();
+
+    int lpi = out.lpi();
+    GAPI_DbgAssert(y + lpi <= outSz.height);
+
+    const auto rowFunc = type_dispatch<resizeArea_suptypes>(in.meta().depth,
+                                                            cv_type_id{},
+                                                            typed_resizeArea<isa_tag_t, T, alpha_type, index_type, work_type>{},
+                                                            nullptr);
+    GAPI_DbgAssert(rowFunc);
 
-    static void run(const cv::gapi::fluid::View& in, Size /*sz*/, int /*interp*/,
-                    cv::gapi::fluid::Buffer& out, cv::gapi::fluid::Buffer &scratch) {
-        calcRowLinear<float, areaUpscale32f::Mapper>(in, out, scratch);
+    for (int l = 0; l < lpi; l++) {
+        Unit ymap = ymapper.map(y + l);
+
+        GAPI_Assert(ymap.index1 - ymap.index0 <= 32);
```

Comment:
could you please give a name to 32 constant?

Reply:
Done.